client/taskenv/env.go

```diff
@@ -98,6 +98,10 @@ const (
 
 	HostAddrPrefix = "NOMAD_HOST_ADDR_"
 
+	// UnixAddr is the task api unix socket, in the appropriate format
+	// for use in a NOMAD_ADDR (i.e. prefixed with "unix:")
+	UnixAddr = "NOMAD_UNIX_ADDR"
```

Comment:
Could this be `NOMAD_ADDR`? That would let workloads that have access to the Nomad binary (or the Nomad `api` package) automatically pick up the right environment variable needed to talk to Nomad, without much in the way of magic. I'm pretty sure it would be overwritten by a `template.env` or `env` block (although we should double-check), which means we would break backcompat either.

Reply:
I would love that, but I have vague fears that folks may have already gotten that in their environment another way, and I don't want to overwrite it.  For example, if they have `NOMAD_ADDR` in their _client agent_ environment, then it would be in a `raw_exec` task's env, but not uncomplicated to check here.  `os.Getenv` is simple, but "it's this, unless X, except for Y, but definitely when Z" seemed more complicated to look for and explain.

Reply:
> For example, if they have NOMAD_ADDR in their client agent environment, then it would be in a raw_exec task's env,

Good shout. I missed `raw_exec`, which is funny because I only just recently reviewed https://github.com/hashicorp/nomad/pull/25511